docs/writing-docs/doc-blocks.md

```diff
@@ -0,0 +1,370 @@
+
+<div style="background-color:#F8FAFC">
+TODO: ask tom about this and how we can improve.
+
+My read on this is that we come from null that shows no control. But we have some shorthands that help the reader to show something.
+
+Probably we could add this as a aside, or give it more emphasis
+</div>
+
+> Storybook provides shorthand for common tasks:
+> type: 'number' is shorthand for type: { name: 'number' }
+> control: 'radio' is shorthand for control: { type: 'radio' }
+
+##### MDX
+
+To customize argTypes in MDX, you can set an `mdx` prop on the `Meta` or `Story` components:
+
+```js
+
+<Meta
+  title="MyComponent"
+  component={MyComponent}
+  argTypes={{
+    label: { 
+        name: 'label',
+        /* other argtypes required */ 
+    },
+  }}
+/>
+
+<Story name="some story" argTypes={{
+  label: { 
+      name: 'different label', 
+      /* other required data */  
+    }
+}}>
+  {/* story contents */}
+</Story>
+```
+
+#### Controls
+
+The controls inside a props table are configured in exactly the same way as the [controls](../essentials/controls) addon pane. In fact you’ll probably notice the table is very similar! It uses the same component and mechanism behind the scenes.
+
+### Source
+
+Storybook Docs displays a story’s source code using the `Source` block. The snippet has built-in syntax highlighting and can be copied with the click of a button.
+
+<div style="background-color:#F8FAFC">
+TODO:add image per screenshot requirements (mention of a image in the SB 6.0 doc (needs to be vetted))
+</div>
+
+#### DocsPage
+
+In DocsPage, the `Source` block appears automatically within each story’s [Preview] block.
+
+<div style="background-color:#F8FAFC">
+TODO:add image per screenshot requirements (mention of a image in the SB 6.0 doc (needs to be vetted))
+</div>
+
+To customize the source snippet that’s displayed for a story, set the `docs.source.code` parameter:
+
+```js
+
+export const CustomSource = () => Template.bind({});
+
+CustomSource.parameters = {
+  docs: { 
+      source: { 
+          code: 'Some custom string here';  
+      } 
+    },
+};
+```
+
+#### MDX
+
+You can also use the `Source` block in MDX. It accepts either a story ID or `code` snippet the `language` for syntax highlighting:
+
+```js
+import { Source } from '@storybook/addon-docs/blocks';
+Import dedent from 'ts-dedent';
+
+<Source
+  language='css'
+  code={dedent`
+     .container {
+       display: grid | inline-grid;
+     }
+  `}
+/>
+```
+
+#### ⚒️ Description
+
+Storybook Docs shows a component’s description extracted from the source code or based on a user-provided string.
+
+<div style="background-color:#F8FAFC">
+TODO:add image per screenshot requirements (mention of a image in the SB 6.0 doc (needs to be vetted))
+</div>
+
+#### DocsPage
+
+<div style="background-color:#F8FAFC">
+TODO: left it for now ask feedback tom/shillman on this based on [pr](https://github.com/storybookjs/storybook/issues/11441)
```

Comment:
@shilman what's up with that PR? https://github.com/storybookjs/storybook/issues/11441

Reply:
Not implemented. I'll add it this week.